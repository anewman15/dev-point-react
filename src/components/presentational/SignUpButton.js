import { Link } from 'react-router-dom';

function SignUpButton() {
  return (
    <div className="navbar-item">
      <Link to="/sign_up" className="button is-light"><strong>Sign up</strong></Link>
    </div>
  );
}

export default SignUpButton;
