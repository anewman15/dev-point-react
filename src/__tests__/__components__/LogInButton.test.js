import { unmountComponentAtNode } from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import SignUpButton from '../../components/presentational/SignUpButton';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('SignUpButton', () => {
  it('renders the SIgnUpButton component to the dom', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <SignUpButton />
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
