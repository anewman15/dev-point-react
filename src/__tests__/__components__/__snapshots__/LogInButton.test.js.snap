// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpButton renders the SIgnUpButton component to the dom 1`] = `
<div
  className="navbar-item"
>
  <a
    className="button is-light is-fullwidth"
    href="/sign_up"
    onClick={[Function]}
  >
    <strong>
      Sign up
    </strong>
  </a>
</div>
`;
